tests/unit/alloc_test.cc

```diff
@@ -331,35 +331,114 @@ SEASTAR_TEST_CASE(test_diagnostics_allocation) {
+    *ret = 'c'; // to prevent compiler from considering this a dead allocation and optimizing it out
+    return ret;
+}
+
+SEASTAR_TEST_CASE(test_sampled_profile_collection_small)
 {
-    BOOST_REQUIRE(!seastar::memory::get_heap_profiling_enabled());
+    {
+        auto stats = seastar::memory::memory_profile();
+        BOOST_REQUIRE_EQUAL(stats.size(), 0);
+    }
+
+    std::size_t count = 100;
+    std::vector<volatile char*> ptrs(count);
+
     seastar::memory::set_heap_profiling_enabled(true);
-    BOOST_REQUIRE(seastar::memory::get_heap_profiling_enabled());
+
+#ifdef __clang__
+    #pragma nounroll
+#endif
+    for (std::size_t i = 0; i < count / 2; ++i) {
+        ptrs[i] = malloc_wrapper(10);
+    }
+
+#ifdef __clang__
+    #pragma nounroll
+#endif
+    for (std::size_t i = count / 2; i < count; ++i) {
+        ptrs[i] = malloc_wrapper(10);
+    }
+
+    // NB: the test framework allocates
+    seastar::memory::set_heap_profiling_enabled(0);
 
     {
         auto stats = seastar::memory::memory_profile();
-        BOOST_REQUIRE_EQUAL(stats.size(), 0);
+
+        BOOST_REQUIRE_EQUAL(stats.size(), 2);
+        BOOST_REQUIRE_EQUAL(stats[0].size, stats[0].count * 100);
+    }
+
+    seastar::memory::set_heap_profiling_enabled(false);
+
+    for (auto ptr : ptrs) {
+        free((void*)ptr);
     }
 
-    volatile char* ptr = static_cast<char*>(malloc(10));
-    *ptr = 'c'; // to prevent compiler from considering this a dead allocation and optimizing it out
+    seastar::memory::set_heap_profiling_enabled(0);
 
     {
         auto stats = seastar::memory::memory_profile();
-        BOOST_REQUIRE_EQUAL(stats.size(), 1);
-        BOOST_REQUIRE_EQUAL(stats[0].size, 32); // 10 + 8 falls into 32 byte pool
+        BOOST_REQUIRE_EQUAL(stats.size(), 0);
     }
 
-    free((void*)ptr);
+    return seastar::make_ready_future();
+}
 
+SEASTAR_TEST_CASE(test_sampled_profile_collection_large)
+{
     {
         auto stats = seastar::memory::memory_profile();
         BOOST_REQUIRE_EQUAL(stats.size(), 0);
     }
 
-    // Needed for now because we can't differentiate between sampled allocations and non-sampled ones
-    seastar::memory::set_heap_profiling_enabled(false);
+    std::size_t count = 100;
+    std::vector<volatile char*> ptrs(count);
+
+    seastar::memory::set_heap_profiling_enabled(1000000);
+
+#ifdef __clang__
+    #pragma nounroll
+#endif
+    for (std::size_t i = 0; i < count / 2; ++i) {
+        ptrs[i] = malloc_wrapper(100000);
+    }
+
+#ifdef __clang__
+    #pragma nounroll
+#endif
+    for (std::size_t i = count / 2; i < count; ++i) {
+        ptrs[i] = malloc_wrapper(100000);
+    }
+
+    // NB: the test framework allocate
+    seastar::memory::set_heap_profiling_enabled(0);
+
+    {
+        auto stats = seastar::memory::memory_profile();
+        BOOST_REQUIRE_EQUAL(stats.size(), 2);
+        BOOST_REQUIRE_EQUAL(stats[0].size, stats[0].count * 1000000);
+    }
+
+    seastar::memory::set_heap_profiling_enabled(1000000);
```

Comment:
Good eye. I accept your offer to ignore this nitpick however as rewriting it to fix this will take a few minutes I could spend doing something like getting a the sesatar codebase auto-formatted. However if there is a concern about bisectability I can fix it.

Reply:
It probably is a bisectability concern, yes. But then again, if we really cared about bisectability, we would make the CI test all commits. Who knows how many bad commits we have out there.

@avikivity How much do we care about the bisectability here?

Reply:
@michoecho - in the force push https://github.com/scylladb/seastar/commit/a7d7f691117f8971d240937bb789cd3e9d8c87ed I have re-written this commit to move the test changes into a later commit and checked that the re-written commits all compile and pass the alloc test.